### PR TITLE
Add fan platform to knx

### PR DESCRIPTION
**Pull request by crazyfx1**
## Proposed change

Add fan platform to knx integration. Knx fans can be controlled by percentage or steps and optionally support oscillation.
The implementation is already using the new fan percentage methods.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/46161
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/knx.markdown [hunk 1]
```diff
@@ -697,6 +699,51 @@ device_class:
   type: string
 {% endconfiguration %}
 
+## Fan
+
+The `knx fan` integration is used to control KNX fans. Following control types are supported:
+
+- Percentage controlled: Fans which set the percentage directly from 0-100%.
+- Step controlled: Fans which have a fixed amount of steps to set. The integration will convert percentage to step automatically. The `max_step` attribute is set to the amount of steps of the fan, not counting the `off`-step. Example: A fan supports the steps 0 to 3. To use this fan the `max_step` attribute has to be set to `3`. The integration will convert the percentage `66 %` to the step `2` when sending data to KNX.
```
Comment (frenck) [inline source/_integrations/knx.markdown]:
```suggestion
The KNX fan integration is used to control KNX fans. Following control types are supported:

- Percentage controlled: Fans which set the percentage directly from 0-100%.
- Step controlled: Fans that have a fixed amount of steps to set. The integration will convert a percentage to step automatically. The `max_step` attribute is set to the number of steps of the fan, not counting the `off`-step. Example: A fan supports the steps 0 to 3. To use this fan the `max_step` attribute has to be set to `3`. The integration will convert the percentage `66 %` to step `2` when sending data to KNX.
```

File: source/_integrations/knx.markdown [hunk 2]
```diff
@@ -697,6 +699,51 @@ device_class:
...
+
+To use your KNX fan in your installation, add the following lines to your top level [KNX Integration](/integrations/knx) configuration key in `configuration.yaml`:
+
+```yaml
+# Example configuration.yaml entry
+knx:
+  fan:
+    - name: 'ceiling fan'
+      address: '9/0/1'
+      state_address: '9/0/2'
```
Comment (frenck) [inline source/_integrations/knx.markdown]:
```suggestion
  fan:
    - name: "ceiling fan"
      address: "9/0/1"
      state_address: "9/0/2"
```

**Review by frenck — CHANGES_REQUESTED** [review]
Thanks, @crazyfx1 👍 

A couple of small changes and it is good to go!

**Comment by crazyfx1** [comment]
Thanks. I applied the requested changes to the whole knx documentation ;)

**Review by frenck — APPROVED** [review]
Thanks, @crazyfx1 👍

**Pull request merged**